.eslintrc.js

```diff
@@ -11,6 +14,46 @@ module.exports = {
     'src/govuk/vendor/polyfills/**/*'
   ],
   overrides: [
+    {
+      extends: 'plugin:jsdoc/recommended',
+      files: ['**/*.{cjs,js,mjs}'],
+      plugins: [
+        'jsdoc'
+      ],
+      rules: {
+        // JSDoc blocks are optional
+        'jsdoc/require-jsdoc': 'off',
+        'jsdoc/require-param-description': 'off',
+        'jsdoc/require-param': 'off',
+
+        // Require hyphens before param description
+        // Aligns with TSDoc style: https://tsdoc.org/pages/tags/param/
+        'jsdoc/require-hyphen-before-param-description': 'warn',
+
+        // Check for valid formatting
+        'jsdoc/check-line-alignment': 'warn',
+        'jsdoc/check-syntax': 'error',
+
+        // Add unknown @jest-environment tag name
+        'jsdoc/check-tag-names': [
+          'warn', {
+            definedTags: ['jest-environment']
+          }
+        ],
+
+        // Add missing .querySelectorAll() type
+        'jsdoc/no-undefined-types': [
+          'error', {
+            definedTypes: ['NodeListOf']
```

Comment:
@romaricpascal Yeah exactly

Without this change then ~`NodeList<Element>`~ `NodeListOf<Element>` would be invalid

One of the JSDoc plugin dependencies must not be aware of it as a known DOM type (possibly because the TypeScript flavour is `NodeListOf` not `NodeList`. Haven't managed to track down exactly where to log this yet.

In Visual Studio Code if you mouse over `.querySelectorAll()` you'll see `NodeListOf<Element>`

<img width="789" alt="Screenshot of querySelectorAll being typed as NodeListOf" src="https://user-images.githubusercontent.com/415517/195412440-0432b551-8f04-4d96-b1b5-f78b638d0280.png">